config/v1/types_infrastructure.go

```diff
@@ -1340,13 +1340,14 @@ type VSpherePlatformNodeNetworking struct {
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.ingressIPs) || has(self.ingressIPs)",message="ingressIPs list is required once set"
 type VSpherePlatformSpec struct {
 	// vcenters holds the connection details for services to communicate with vCenter.
-	// Currently, only a single vCenter is supported.
+	// Currently, only a single vCenter is supported, but in tech preview 3 vCenters are supported.
 	// ---
 	// + If VCenters is not defined use the existing cloud-config configmap defined
 	// + in openshift-config.
 	// +kubebuilder:validation:MinItems=0
 	// +openshift:validation:FeatureGateAwareMaxItems:featureGate="",maxItems=1
 	// +openshift:validation:FeatureGateAwareMaxItems:featureGate=VSphereMultiVCenters,maxItems=3
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf || size(self) == size(oldSelf) || (size(self) == 1 && size(oldSelf) == 0)",message="vCenters cannot be added or removed once set"
```

Comment:
Does the first clause here matter? Do we care if they are identical or can we just validate the size changes?

Reply:
sorry, this was an older push.  I've been stuck testing a scenario where the vsphere section did not exist.  The new push should have the adjusted rule.